src/modules/mc_pos_control/PositionControl/ControlMathTest.cpp

```diff
@@ -0,0 +1,181 @@
+/****************************************************************************
+ *
+ *   Copyright (C) 2019 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include <gtest/gtest.h>
+#include <ControlMath.hpp>
+#include <px4_defines.h>
+
+using namespace matrix;
+using namespace ControlMath;
+
+
+TEST(ControlMathTest, ThrottleAttitudeMapping)
+{
+	/* expected: zero roll, zero pitch, zero yaw, full thr mag
+	 * reason: thrust pointing full upward */
+	Vector3f thr{0.0f, 0.0f, -1.0f};
+	float yaw = 0.0f;
+	vehicle_attitude_setpoint_s att{};
+	att = thrustToAttitude(thr, yaw);
+	EXPECT_EQ(att.roll_body, 0);
+	EXPECT_EQ(att.pitch_body, 0);
+	EXPECT_EQ(att.yaw_body, 0);
+	EXPECT_EQ(att.thrust_body[2], -1.f);
+
+	/* expected: same as before but with 90 yaw
+	 * reason: only yaw changed */
+	yaw = M_PI_2_F;
+	att = thrustToAttitude(thr, yaw);
+	EXPECT_EQ(att.roll_body, 0);
+	EXPECT_EQ(att.pitch_body, 0);
+	EXPECT_EQ(att.yaw_body, M_PI_2_F);
+	EXPECT_EQ(att.thrust_body[2], -1.f);
+
+	/* expected: same as before but roll 180
+	 * reason: thrust points straight down and order Euler
+	 * order is: 1. roll, 2. pitch, 3. yaw */
+	thr = Vector3f(0.0f, 0.0f, 1.0f);
+	att = thrustToAttitude(thr, yaw);
+	EXPECT_NEAR(abs(att.roll_body), M_PI_F, 1e4f);
```

Comment:
3.14 +/- 10'000 radians, that's quite easy to pass :D

Reply:
The test was wrong before because the absolute value parenteses were not over the difference: https://github.com/PX4/Firmware/pull/13247/files#diff-cda7454ea101ac3644667d5147ee77cdL96

And I managed to break it in a different way :man_facepalming:

Reply:
Ah now I remember: The Mac inaccuracy of the implementation showing up in CI was the problem I added a threshold in the first place: http://ci.px4.io:8080/blue/organizations/jenkins/PX4_misc%2FFirmware-compile_mac/detail/PR-13247/2/pipeline
Let me fix it.